ir/pointer.cpp

```diff
@@ -292,10 +292,14 @@ Pointer Pointer::maskOffset(const expr &mask) const {
            getAttrs() };
 }
 
-expr Pointer::addNoOverflow(const expr &offset) const {
+expr Pointer::addNoSOverflow(const expr &offset) const {
   return getOffset().add_no_soverflow(offset);
 }
 
+expr Pointer::addNoUOverflow(const expr &offset) const {
+  return getAddress().zextOrTrunc(offset.bits()).add_no_uoverflow(offset);
```

Comment:
I think you want something like:
return getAddress().add_no_uoverflow(offset.sextOrTrunc(bits_ptr_address));

An address is never smaller than offsets. We try to shrink the number of bits of offsets as an optimization, hence they are usually not 64 bit long.
(I don't have a fresh build of LLVM to try this out -- maybe in half an hour)